From 1bb1fd8ac4ebf8b7efd6af75e28e1bdd91f25502 Mon Sep 17 00:00:00 2001
From: Niko Tyni <ntyni@debian.org>
Date: Mon, 4 Aug 2014 17:21:17 +0300
Subject: [PATCH] Disable the tree-dse gcc optimization for
 modperl_env_table_populate()

This is a workaround for breakage with gcc 4.9.

Bug-Debian: https://bugs.debian.org/754901
---
 src/modules/perl/modperl_env.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/modules/perl/modperl_env.c b/src/modules/perl/modperl_env.c
index e700501..78d3e44 100644
--- a/src/modules/perl/modperl_env.c
+++ b/src/modules/perl/modperl_env.c
@@ -112,6 +112,8 @@ void modperl_env_clear(pTHX)
 #define MP_ENV_HV_STORE_TABLE_ENTRY(hv, elt)    \
     MP_ENV_HV_STORE(hv, elt.key, elt.val);
 
+/* see https://bugs.debian.org/754901 */
+__attribute__((optimize("-fno-tree-dse")))
 static void modperl_env_table_populate(pTHX_ apr_table_t *table)
 {
     HV *hv = ENVHV;
-- 
2.0.1

